### ISSUE TITLE
Variadic mapping

### ISSUE THREAD

**Issue by lionelgit** [hadley/lowliner#29]
`map2()` and `map3()` are nice shortcuts for specific cases but it would be good to have a way of mapping functions with a variable number of arguments.

We could have for example

* `map_dots(..., .f)`
* `map_list(.l, .f, ...)`

The former would be translated to `Map(.f, ...)` while the latter would be equivalent to `splat(map_dots)(c(.l, list(..., .f = .f)))`.

**Comment by hadley** [comment]
The problem is that to accept an arbitrary number of arguments you either have to do some magic (i.e. assume that the function is the last thing in ...) or require it to be named. Neither approach seems that nice to me, given how rarely you need to map four or more lists simultaneously. 

(I originally thought being explicit by adding the number of arguments to the function name was icky, but that's what Haskell does, and I've grown to like how explicit it is. Plus I think it's easier for most people to understand - `base::Map()` is hard for most people to grasp)

**Comment by lionelgit** [comment]
> you either have to do some magic or require it to be named.

The third option is to require the arguments to be encapsulated in a list, as in `map_list()` above (and as in `cross()`). This way we have a pipe-y interface that blends well in lowliner.

However `map_list()` is probably even harder to understand than `Map()`. But I find it sad that R has such heterogeneous interfaces and naming conventions, so I'd be happy to avoid using `Map()` for the 1% of use cases for which `map2()` and `map3()` are not enough.

**Comment by hadley** [comment]
Yeah, and `list()` isn't so bad memory wise now that it doesn't make a deep copy

**Comment by hadley** [comment]
How about 

```R
map_n <- function(.l, .f, ...) {
  do.call("Map", c(list(.f), .l)) %>% output_hook(.l)
}

n_args <- function(...) nargs()
list(x = 1:3, y = 3:1, z = 2:4) %>%
  map_n(n_args)
```

I think I'd prefer to stay away from ...

**Comment by lionelgit** [comment]
Closed by #27
